Add documentation about the platform module (#69)
micronaut-projects/micronaut-platform: src/main/docs/guide/introduction.adoc

> This Micronaut module specifies the Bill of Materials (BOM) for the Micronaut Project, also known at the Micronaut Platform.

Reviewer: ```suggestion
This Micronaut module specifies the Bill of Materials (BOM) for the Micronaut Project, also known as the Micronaut Platform.
```

The Micronaut Platform is a collection of modules that can be used to build Micronaut applications and have been verified to work well together.

It aggregates many of the individual Micronaut module BOMs and provides a single BOM to manage the versions of all the modules.

> If you are using either the https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/[Micronaut Gradle Plugin] or the https://micronaut-projects.github.io/micronaut-maven-plugin/latest/[Micronaut Maven Plugin], you do not need to add this module to your project as it will be added automatically.

Reviewer: ```suggestion
If you use either the https://micronaut-projects.github.io/micronaut-gradle-plugin/latest/[Micronaut Gradle Plugin] or the https://micronaut-projects.github.io/micronaut-maven-plugin/latest/[Micronaut Maven Plugin], you do not need to add this module to your project as it will be added automatically.
```

> NOTE: The Micronaut Platform BOM exists since Micronaut 4, and corresponds to the Micronaut 3 BOM which was published at `io.micronaut:micronaut-bom`. In Micronaut 4, we provide both the Micronaut Platform BOM (`io.micronaut.platform:micronaut-platform`) and the Micronaut Core BOM (`io.micronaut:micronaut-core-bom`).

Reviewer: ```suggestion
NOTE: The Micronaut Platform BOM exists since Micronaut Framework 4, and corresponds to the Micronaut Framework  3 BOM which was published at `io.micronaut:micronaut-bom`. In Micronaut 4, we provide both the Micronaut Platform BOM (`io.micronaut.platform:micronaut-platform`) and the Micronaut Core BOM (`io.micronaut:micronaut-core-bom`).
```

Using Micronaut as a noun is against the brand guidelines.

== The Micronaut Platform catalog

> For Gradle users, the platform is also published as a version catalog which can be used in your build scripts to easily add "blessed" dependencies to your project.

Reviewer: ```suggestion
For Gradle users, the platform is also published as a https://docs.gradle.org/current/userguide/platforms.html#sub:version-catalog[version catalog] which can be used in your build scripts to add "blessed" dependencies to your project.
```

In order to use the version catalog, import it into your `settings.gradle(.kts)` file:

> [source,groovy]

Reviewer: ```suggestion
[source,kotlin]
```

----
dependencyResolutionManagement {
    repositories {
        mavenCentral()
    }
    versionCatalogs {
        create("mn") {
            from("io.micronaut:micronaut-platform:${micronautVersion}")
        }
    }
}
----

Then, you can use the catalog to add dependencies to your project:

> [source,groovy]

Reviewer: ```suggestion
[source,kotlin]
```

----

> implementation mn.micronaut.data.jdbc

Reviewer: ```suggestion
implementation(mn.micronaut.data.jdbc)
```

Use a syntax valid for Kotlin and Groovy DSL.
